### PR TITLE
Implemented lshift

### PR DESCRIPTION
**Pull request by Kishan-Ved**
Fixes: https://github.com/lfortran/lfortran/issues/3137

As per the documentation of lshift: https://gcc.gnu.org/onlinedocs/gcc-4.4.7/gfortran/LSHIFT.html

### CODE REVIEW

**Comment by Kishan-Ved** [comment]
@HarshitaKalani this looks good?

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -1755,6 +1757,75 @@ namespace Shiftr {
...
+        }
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::Lshift),
+            args.p, args.n, 0, ASRUtils::expr_type(args[0]), m_value);
+    }
+
+    static inline ASR::expr_t* instantiate_Lshift(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_sign_" + type_to_str_python(arg_types[0]));
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_function_registry.h:1809]:
```suggestion
        declare_basic_variables("_lcompilers_lshift_" + type_to_str_python(arg_types[0]));
```

File: integration_tests/intrinsics_123.f90 [hunk 2]
```diff
@@ -0,0 +1,18 @@
+program intrinsics_123
+
+    print*, lshift(10_4, 0)
+    if (.not. shiftr(10_4, 0) == 10) error stop
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_123.f90:4]:
```suggestion
    if (.not. lshift(10_4, 0) == 10) error stop
```

**Comment by Kishan-Ved** [comment]
Shiftl implementation has been merged, resulting in conflicts here. Do we need lshift? (Then I'll resolve the conflicts.)

**Comment by Pranavchiku** [comment]
> Do we need lshift? (Then I'll resolve the conflicts.)

We have lshift working in main with PR #3164. Closing with reference to it.

**Pull request closed without merging**